### PR TITLE
Fix macros missing KOKKOS_ prefix, unnecessarily left defined

### PR DESCRIPTION
**Pull request by PhilMiller**

### CODE REVIEW

File: core/src/Kokkos_Crs.hpp [hunk 1]
```diff
@@ -304,11 +304,11 @@ struct CountAndFillBase {
   Functor m_functor;
   counts_type m_counts;
   struct Count {};
-  inline void operator()(Count, size_type i) const {
+  KOKKOS_INLINE_FUNCTION void operator()(Count, size_type i) const {
```
Comment (masterleinad) [inline core/src/Kokkos_Crs.hpp:307]: In the spirit of removing `inline` from functions being defined in the class declaration.
```suggestion
  KOKKOS_FUNCTION void operator()(Count, size_type i) const {
```

File: core/src/Kokkos_Crs.hpp [hunk 2]
```diff
@@ -304,11 +304,11 @@ struct CountAndFillBase {
...
     m_counts(i) = m_functor(i, nullptr);
   }
   struct Fill {};
-  inline void operator()(Fill, size_type i) const {
+  KOKKOS_INLINE_FUNCTION void operator()(Fill, size_type i) const {
```
Comment (masterleinad) [inline core/src/Kokkos_Crs.hpp:311]:
```suggestion
  KOKKOS_FUNCTION void operator()(Fill, size_type i) const {
```

File: core/src/Kokkos_Crs.hpp (hunk 1 above)
Comment (PhilMiller) [reply core/src/Kokkos_Crs.hpp:307]: I've probably missed some memo or discussion, but why is that preferable, or what's the guideline?

Comment (dalg24) [reply core/src/Kokkos_Crs.hpp:307]: Do not inline function that are defined in the body of a class

Comment (PhilMiller) [reply core/src/Kokkos_Crs.hpp:307]: Ok, will revise

File: core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp [hunk 3]
```diff
@@ -62,33 +62,37 @@ template <class ValueType, class ReducerType>
 __device__ inline void cuda_inter_warp_reduction(
     ValueType& value, const ReducerType& reducer,
     const int max_active_thread = blockDim.y) {
-#define STEP_WIDTH 4
+#define KOKKOS_IMPL_STEP_WIDTH 4
```
Comment (masterleinad) [inline core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp:65]: I don't see a good reason to use a macro here in the first place.
```suggestion
constexpr int step_width = 4;
```

Comment (PhilMiller) [reply core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp]: Changed.

**Comment by PhilMiller** [comment]
This rebases cleanly onto `release-candidate-4.0` if we want it there.

**Review by masterleinad — APPROVED** [review]

**Comment by rgayatri23** [comment]
LGTM for OpenMPTarget stuff.

**Review by rgayatri23 — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]
For the record Bruno had written on nucleus in the discussion that preceded that PR

> HIP_ENABLE_PRINTF can be removed. It was necessary for older versions of the compiler

**Comment by crtrott** [comment]
The failure is with random test so its likely fine.

**Review by crtrott — APPROVED** [review]

**Comment by PhilMiller** [comment]
Do you want me to post the 4.0 PR as well?

**Comment by dalg24** [comment]
Sure

**Comment by PhilMiller** [comment]
4.0 cherry-pick posted in #5842.

**Pull request merged**